src/test-utils/dom/date-range-picker/index.ts

```diff
@@ -122,7 +123,7 @@ export class DrpDropdownWrapper extends ComponentWrapper {
   ): CalendarDateWrapper {
     const gridClassName = grid === 'right' ? styles['second-grid'] : styles['first-grid'];
     return this.findComponent(
-      `.${gridClassName} .${gridStyles.week}:nth-child(${row}) .${gridStyles.day}:nth-child(${column})`,
+      `.${gridClassName} .${testStyles['calendar-week']}[data-awsui-weekindex="${row}"] .${testStyles['calendar-date']}:nth-child(${column})`,
```

Comment:
Using data- attribute instead of :nth-child() to exclude rows with placeholder dates only. That is necessary for the backward-compatibility, to ensure the indices of the monthly dates do not change.